[Ong Xin Bin] iP

n3wsoldier opened the pull request.

n3wsoldier committed:
Update Duke.java

Implement an initial skeletal version of the Duke that simply greets the user and exits.

n3wsoldier committed:
Update Duke.java: echo commands

1) Implement user input and echos commands.
2) User input "bye" exit program

n3wsoldier committed:
Update Duke.java: store input

1) Store input by the user
2) Add "list" condition check to list stored inputs

n3wsoldier committed:
Add Task.java: Task Class

Task class:
1) Constructor to with string input to store the description
2) Static numberOfTask to keep track of tasks stored
3) getStatusIcon(): return a tick or cross based on isDone
4) markAsDone(): change isDone to true
5) getDescription(): return description
6) getNumberOfTask(): return the total number task

n3wsoldier committed:
Update Duke.java: mark as done

1) Replace string array with the Task array
2) Additional check for "done " to mark task as done
3) Using task class to access attribute

n3wsoldier committed:
Add Type.java: Type Enum

1) Todo
2) Deadline
3) Event

n3wsoldier committed:
Update Task.java: Store Type

1) Add type variable
2)Update constructor to store variable
3) Create @overide toString method

n3wsoldier committed:
Add Todo.java: Todo class

1) Extends Task
2) Constructor using Task(super)

n3wsoldier committed:
Add Deadline.java: Deadline class

1) Extends Task
2) Constuctor using Task
3) tostring to overide add by:

n3wsoldier committed:
Add Event.java: event class

1) Extends task
2) Constructor using task (super)

n3wsoldier committed:
Update Duke.java: adding new command

1) Using the new command to add specific class
2) Use task tostring function
3) Print task added message function

n3wsoldier committed:
Update runtest.bat: change command

n3wsoldier committed:
Refactor Duke.java

n3wsoldier committed:
Update Duke.java: adding description for method

chocomango commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+        String[] inputs = userCommand.split(" ", PARAM_DELIMIT_LIMIT);
+        String command = inputs[0];
+        switch (command){
+        case COMMAND_LIST:
```
Good indentation for the the switch statement.

EthanWong2212 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+    private static final String PARAM_DELIMIT_AT = "/at";
+    private static final int PARAM_DELIMIT_LIMIT = 2;
+
+    private static Task[] tasks = new Task[100];
```
Perhaps you can refractor the number 100 to show the meaning of it since its a constant?

EthanWong2212 reviewed (commented):
good job

Wu-Haitao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+            + "|  _ \\ _   _| | _____ "+LS
+            + "| | | | | | | |/ / _ \\"+LS
+            + "| |_| | |_| |   <  __/"+LS
+            + "|____/ \\__,_|_|\\_\\___|"+LS;
```
Personally I prefer to align these lines in order to show the full logo completely. And it would be better if you put spaces before and after the "+".

Wu-Haitao commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+    private static void listTasks(){
+        System.out.println(MESSAGE_LINE);
+        System.out.println(MESSAGE_LIST);
+        for(int i = 0; i < Task.getNumberOfTasks(); i++){
```
Missing a space here.
It seems that you missed every spaces after the first lines of the methods.

Wu-Haitao reviewed (commented):
Generally good. Just try polish in detail.

n3wsoldier committed:
Update Duke.java: error handling

1) Error handling
2) print error function

n3wsoldier committed:
Merge branch 'branch-Level-5'

n3wsoldier committed:
Create task package

n3wsoldier committed:
Merge branch 'branch-A-Packages'

n3wsoldier committed:
Add DukeException.java

n3wsoldier committed:
Merge branch 'branch-Level-5'

n3wsoldier committed:
Create duke package

n3wsoldier committed:
Merge branch 'branch-A-Packages'

chocomango commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,241 @@
+    private static final String PARAM_DELIMIT_AT = " /at ";
+    private static final int PARAM_DELIMIT_LIMIT = 2;
+
+    private static Task[] tasks = new Task[100];
```
I like how all the constants are named and used in the code. Really improved the code's readability.

chocomango commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,241 @@
+import java.util.Scanner;
+
+public class Duke {
+    private static final String LS = System.lineSeparator();
```
Just curious if the name _LS_ would be confusing as NEWLINE seems to be more intuitive.

chocomango commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+    public Deadline(String description, String by){
+        super(description, TaskType.Deadline);
+        this.by = by;
+        this.toString();
```
Is there a reason to include _this.toString()_ in the constructor?

chocomango commented on src/main/java/duke/task/TaskType.java:
```diff
@@ -0,0 +1,5 @@
+package duke.task;
+
+public enum TaskType {
+    Event, Deadline, Todo
```
Since this enum is limited to the Task class, it may be better to declare it in the class to limit its scope.
Also, since enum is a group of constants, it is better to use all uppercase.

chocomango reviewed (changes requested):
Really liked how the methods are used in the code, making it easy to read and edit. Overall, good job and keep up the work.

n3wsoldier committed:
Update Duke.java: delete task

n3wsoldier committed:
Update Duke.java: save task list

n3wsoldier committed:
Merge branch 'branch-Level-6'

n3wsoldier committed:
Merge branch 'branch-Level-7'

n3wsoldier committed:
Update Duke.java: User interaction/ help command

n3wsoldier committed:
Update Duke.java: help menu

n3wsoldier committed:
MoreOOP: Spliting Duke.java into relavent packages and subclass

1) Creating relavent packages
2) Creating relavent classes and refactore the proper function into those classe
3) Adding proper comments for all function

Samuel787 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    /***
+     * Looped to read usercommand
```
You may want to consider using present tense for comments

Samuel787 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,38 @@
+
+    /***
+     * DeadlineCommand constructor using string values.
+     * @param description : description of task
```
You may consider inserting an empty line between method description and params.
https://se-education.org/guides/conventions/java/basic.html#java-coding-standard-basic

Samuel787 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,35 @@
+    private int toDelete;
+
+    /***
+     * DoneCommand constructor using int values.
```
Comments mention DoneCommand, however, the constructor here is DeleteCommand

Samuel787 commented on src/main/java/duke/common/Messages.java:
```diff
@@ -0,0 +1,49 @@
+package duke.common;
```
Good job extracting these strings out as constants

Samuel787 commented on src/main/java/duke/data/TaskList.java:
```diff
@@ -0,0 +1,100 @@
+                taskSaveFormat = StorageManager.SYMBOL_EVENT + StorageManager.PARAM_DELIMIT_SAVE + isDone + StorageManager.PARAM_DELIMIT_SAVE + task.getDescription()
+                        + StorageManager.PARAM_DELIMIT_SAVE +((Event) task).getAt();
+            }
+//            System.out.println("\t "+ taskSaveFormat);
```
You may want to remove old code instead  of merely commenting them out before making PR

Samuel787 commented on src/main/java/duke/data/storage/StorageManager.java:
```diff
@@ -0,0 +1,155 @@
+     * @param saveContent : Formated task data item
+     */
+    public void save(String saveContent) {
+//        System.out.println("\t Saving to :\""+ fileName +"\""+ Messages.LS);
```
You may want to remove old code instead of commenting them out before making the PR

Samuel787 commented on src/main/java/duke/data/storage/StorageManager.java:
```diff
@@ -0,0 +1,155 @@
+     * @return Arraylist <Task>
+     */
+    private ArrayList<Task> parse(ArrayList<String> dataItems){
+        //inputs[0] = command
+        //inputs[1] = arguments
```
Old code

Samuel787 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,121 @@
+     */
+    public Command executeCommand(String userCommand) throws InvalidCommandException,
+            DescriptionEmptyException, DueTimeEmptyException{
+        //inputs[0] = command
+        //inputs[1] = arguments
```
Old code

Samuel787 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,121 @@
+
+public class Parser {
+
+    public final int PARAM_DELIMIT_LIMIT = 2;
```
Good job extracting this out as a constant so that it doesn't appear as a magic number.

Samuel787 reviewed (commented):
Generally well done. In some classes, you have forgotten to delete old code and just commented it out. Overall, good coding standards and code quality observed.

n3wsoldier committed:
Add DateTimeValidator.java: interface that convert due time

1) To be implemented by Deadline and Event class
2) Check for valid date format and convert

n3wsoldier committed:
Add FindCommand: find task with searchphrase

1) Check description for certain search phrase

n3wsoldier committed:
Adding JavaDoc comments

n3wsoldier committed:
Merge pull request #1 from n3wsoldier/branch-Level-8

Merge branch 'branch-Level-8'

n3wsoldier committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	data/duke.txt

n3wsoldier committed:
Merge pull request #2 from n3wsoldier/branch-Level-9

Merge branch-Level-9

n3wsoldier committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/data/task/Task.java

n3wsoldier committed:
Merge pull request #3 from n3wsoldier/branch-A-JavaDoc

Merge branch-A-JavaDoc

n3wsoldier committed:
Set theme jekyll-theme-cayman

n3wsoldier committed:
Update UI and Error handling

n3wsoldier committed:
Update Userguide

n3wsoldier committed:
Merge branch 'master' of https://github.com/n3wsoldier/ip

n3wsoldier committed:
Update Userguide

n3wsoldier committed:
Set theme jekyll-theme-leap-day

n3wsoldier committed:
Update README.md

n3wsoldier committed:
Update README.md

n3wsoldier committed:
Set theme jekyll-theme-cayman

n3wsoldier committed:
Update README.md

n3wsoldier committed:
Update README.md

n3wsoldier committed:
Bug fixes

1) Added Number format exception
2) Update help message
3) Userguide added command summary and faq
4) Ui improvement

n3wsoldier committed:
Update Messages

n3wsoldier committed:
Update README.md: Include release link

n3wsoldier committed:
Update README.md: starting output

n3wsoldier committed:
Update README.md: output

n3wsoldier committed:
Update JavaDoc
